internal/node/node_loader.js

```diff
@@ -456,38 +457,77 @@ if (TEMPLATED_install_source_map_support) {
     `);
   }
 }
-// Load all bootstrap modules before loading the entrypoint.
-for (var i = 0; i < BOOTSTRAP.length; i++) {
-  try {
-    module.constructor._load(BOOTSTRAP[i], this);
-  } catch (e) {
-    console.error('bootstrap failure ' + e.stack || e);
-    process.exit(1);
-  }
+
+// Check if we need to profile this run.
+let beforePromise = Promise.resolve();
+let afterPromise = Promise.resolve();
+
+if (PROFILE) {
+  const inspector = require('inspector');
+  const session = new inspector.Session();
+  session.connect();
+
+  // Setup the lifecycle hooks around the execution.
+  beforePromise = new Promise((resolve) => {
+    session.post('Profiler.enable', () => {
+      session.post('Profiler.start', () => {
+        
+        // Set the profile writing logic.
+        afterPromise = () => new Promise((resolve, reject) => {
+          session.post('Profiler.stop', (err, { profile }) => {
+            if (!err) {
+              // TODO: new to decide on the real name/path here. Where should it go?
+              const profilePath = path.resolve(process.cwd(), 'PROFILE_NAME_HERE-' + Math.floor(Math.random() * 99999) + '.cpuprofile');
```

Comment:
Where should these go, and what should they be named?

Reply:
same behavior as `mktemp -d` works for me. It doesn't need to be elegant, just need to be able to find them, right? maybe including the program's name like `$0` would help to locate the right one?